sphinx_gallery/gen_rst.py

```diff
@@ -515,9 +530,14 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
             else:
                 example_rst += text2string(bcontent) + '\n'
 
+    amount_of_code = sum([len(bcontent)
+                          for blabel, bcontent in script_blocks
+                          if blabel == 'code'])
+
```

Comment:
This block is repeated. I would put it only once before the `if _plots_are_current` clause